src/Util/Lock.php

```diff
@@ -21,191 +19,79 @@
  */
 class Lock
 {
-	private static $semaphore = [];
-
 	/**
-	 * @brief Check for memcache and open a connection if configured
-	 *
-	 * @return object|boolean The memcache object - or "false" if not successful
+	 * @var Lock\ILockDriver;
 	 */
-	private static function connectMemcache()
-	{
-		if (!function_exists('memcache_connect')) {
-			return false;
-		}
-
-		if (!Config::get('system', 'memcache')) {
-			return false;
-		}
+	static $driver = null;
 
-		$memcache_host = Config::get('system', 'memcache_host', '127.0.0.1');
-		$memcache_port = Config::get('system', 'memcache_port', 11211);
-
-		$memcache = new Memcache;
-
-		if (!$memcache->connect($memcache_host, $memcache_port)) {
-			return false;
+	public static function init()
+	{
+		switch(Config::get('system', 'lock_driver', 'default')) {
+			case 'memcache':
+				self::$driver = new Lock\MemcacheLockDriver();
```

Comment:
This should check that the Cache driver isn't `database`, otherwise the Database lock driver should be used instead.

Reply:
So we have the options:
- auto
- database
- cache
?

Reply:
And:
- semaphore